pkg/runtime/wfengine/workflowstate.go

```diff
@@ -152,6 +152,19 @@ func addStateOperations(req *actors.TransactionalRequest, keyPrefix string, even
 	return nil
 }
 
+func addPurgeStateOperations(req *actors.TransactionalRequest, keyPrefix string, events []*backend.HistoryEvent) error {
+	// TODO: Investigate whether Dapr state stores put limits on batch sizes. It seems some storage
+	//       providers have limits and we need to know if that impacts this algorithm:
+	//       https://learn.microsoft.com/azure/cosmos-db/nosql/transactional-batch#limitations
```

Comment:
I think we should open a bug to track doing purges in smaller batches to avoid running into transactional batch limitations.

Reply:
https://github.com/dapr/dapr/issues/6289